src/vs/workbench/services/untitled/common/untitledEditorService.ts

```diff
@@ -266,10 +266,10 @@ export class UntitledEditorService extends Disposable implements IUntitledEditor
 		return this.mapResourceToAssociatedFilePath.has(resource);
 	}
 
-	suggestFileName(resource: URI): string | undefined {
+	suggestFileName(resource: URI): string {
 		const input = this.get(resource);
 
-		return input ? input.suggestFileName() : undefined;
+		return input ? input.suggestFileName() : 'untitled';
```

Comment:
@mjbvz I thought rather to return `baseNameOrAuthority(resource)` here, which would take care of e.g. convert a URI of `untitled://Untitled-7` to the name `Untitled-7`